src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -365,6 +366,62 @@ def end_partition(self, pdf: pandas.DataFrame) -> pandas.DataFrame:
 
     applyInPandas = apply_in_pandas
 
+    def pivot(
+        self, pivot_col: ColumnOrName, values: Iterable[LiteralType]
+    ) -> "RelationalGroupedDataFrame":
+        """Rotates this DataFrame by turning unique values from one column in the input
+        expression into multiple columns and aggregating results where required on any
+        remaining column values.
+
+        Only one aggregate is supported with pivot.
+
+        Args:
+            pivot_col: The column or name of the column to use.
+            values: A list of values in the column.
+
+        Example::
+
+            >>> create_result = session.sql('''create or replace temp table monthly_sales(empid int, team text, amount int, month text)
+            ... as select * from values
+            ... (1, 'A', 10000, 'JAN'),
+            ... (1, 'B', 400, 'JAN'),
+            ... (2, 'A', 4500, 'JAN'),
+            ... (2, 'A', 35000, 'JAN'),
+            ... (1, 'B', 5000, 'FEB'),
+            ... (1, 'A', 3000, 'FEB'),
+            ... (2, 'B', 200, 'FEB') ''').collect()
+            >>> df = session.table("monthly_sales")
+            >>> df.group_by("empid").pivot("month", ['JAN', 'FEB']).sum("amount").sort(df["empid"]).show()
+            -------------------------------
+            |"EMPID"  |"'JAN'"  |"'FEB'"  |
+            -------------------------------
+            |1        |10400    |8000     |
+            |2        |39500    |200      |
+            -------------------------------
+            <BLANKLINE>
+
+            >>> df.group_by(["empid", "team"]).pivot("month", ['JAN', 'FEB']).sum("amount").sort("empid", "team").show()
+            ----------------------------------------
+            |"EMPID"  |"TEAM"  |"'JAN'"  |"'FEB'"  |
+            ----------------------------------------
+            |1        |A       |10000    |3000     |
+            |1        |B       |400      |5000     |
+            |2        |A       |39500    |NULL     |
+            |2        |B       |NULL     |200      |
+            ----------------------------------------
+            <BLANKLINE>
+        """
+        if not values:
+            raise ValueError("values cannot be None or empty")
+        pc = self._df._convert_cols_to_exprs(
+            "RelationalGroupedDataFrame.pivot()", pivot_col
+        )
+        value_exprs = [
+            v._expression if isinstance(v, Column) else Literal(v) for v in values
+        ]
+        self._group_type = _PivotType(pc[0], value_exprs)
+        return self
```

Comment:
You could have the return-type be [typing_extensions.Self](https://docs.python.org/3/library/typing.html#typing.Self) here